### PR TITLE
LE Audio: Unicast audio fixes

### LINKED ISSUE #46346: LE Audio: Fatal crash when sending Audio data
**Describe the bug**
The broadcast source and unicast client both fail fatally when attempting to send data on a audio stream. 

**To Reproduce**
Flash and run the broadcast source (as that fails standalone) sample. 

**Expected behavior**
The broadcast source to start sending audio data

**Impact**
Showstopper for LE Audio. The issue seems to persist in both samples (although not failing babblesim). 

**Logs and console output**
```
[00:00:00.591,674] <dbg> bt_iso: hci_le_big_complete: BIG[0] 0x200061e4 completed, status 0
[00:00:00.591,796] <dbg> bt_iso: bt_iso_connected: 0x20005be0
[00:00:00.592,498] <dbg> bt_iso: bt_iso_chan_set_state_debug: chan 0x20006218 iso 0x20005be0 connecting -> connected
[00:00:00.592,559] <dbg> bt_audio_broadcast_source: broadcast_source_iso_connected: stream 0x20006218 ep 0x2000623c
[00:00:00.592,620] <dbg> bt_audio_broadcast_source: broadcast_source_set_ep_state: ep 0x2000623c id 0x00 qos-configured -> streaming
[00:00:00.595,306] <dbg> bt_iso: bt_iso_chan_send: chan 0x20006218 len 40
[00:00:00.595,703] <dbg> bt_iso: bt_iso_chan_send: chan 0x20006218 len 40
[00:00:00.603,057] <dbg> bt_audio_broadcast_source: broadcast_source_iso_sent: stream 0x20006218 ep 0x2000617c
[00:00:00.603,118] <err> os: mem_manage_fault: ***** MPU FAULT *****
[00:00:00.603,179] <err> os: mem_manage_fault:   Instruction Access Violation
[00:00:00.603,271] <err> os: esf_dump: r0/a1:  0x00000000  r1/a2:  0x00000019  r2/a3:  0x00000000
[00:00:00.603,363] <err> os: esf_dump: r3/a4:  0xff4672b5 r12/ip:  0x00000000 r14/lr:  0x0000f9e3
[00:00:00.603,424] <err> os: esf_dump:  xpsr:  0xa1000000
[00:00:00.603,485] <err> os: esf_dump: Faulting instruction address (r15/pc): 0xff4672b4
[00:00:00.603,576] <err> os: z_fatal_error: >>> ZEPHYR FATAL ERROR 0: CPU exception on CPU 0
[00:00:00.603,637] <err> os: z_fatal_error: Current thread: 0x20005e68 (unknown)
[00:00:00.917,205] <err> os: k_sys_fatal_error_handler: Halting system

```

**Environment (please complete the following information):**
 - Commit SHA or Version used: c62ee3be4e4f9c268a009bced3b4d969d7ccbe46

**Additional context**
Basically the same application does not fail as a babblesim test (https://github.com/zephyrproject-rtos/zephyr/pull/46204)

The iso broadcast and central iso samples are also fully functioning, so it's likely either some parameter or misconfiguration in the LE Audio stack.

### LINKED ISSUE #46369: LE Audio: Bidirectional stream is not created
**Describe the bug**
Recent PR merged https://github.com/zephyrproject-rtos/zephyr/pull/42964 supposed to provide support for CIS bidirectional stream.
While testing unicast_audio_client with unicast_audio_server the stream created is not bidirectional, but unidirectional.
Moreover, unicast_audio_client crashes.

**To Reproduce**
1. build the samples and flash on the board
2. Observe the 2 CISes created is unidirectional
3. Observe crash on unicast_audio_client

**Expected behavior**
Bidirectional stream is created, unicast_audio_client don't crash

```
> HCI Event: LE Meta Event (0x3e) plen 7                                                                                                                                                                                                       #152 30.779300
      LE Connected Isochronous Stream Request (0x1a)
        ACL Handle: 0
        CIS Handle: 2
        CIG ID: 0x00
        CIS ID: 0x00
< HCI Command: LE Accept Connected Isochronous Stream Request (0x08|0x0066) plen 2                                                                                                                                                             #153 30.779400
        CIS Handle: 2
> HCI Event: Command Status (0x0f) plen 4                                                                                                                                                                                                      #154 30.779600
      LE Accept Connected Isochronous Stream Request (0x08|0x0066) ncmd 1
        Status: Success (0x00)
> HCI Event: LE Meta Event (0x3e) plen 29                                                                                                                                                                                                      #155 31.010800
      LE Connected Isochronous Stream Established (0x19)
        Status: Success (0x00)
        Connection Handle: 2
        CIG Synchronization Delay: 4380 us (0x00111c)
        CIS Synchronization Delay: 4380 us (0x00111c)
        Master to Slave Latency: 4380 us (0x00111c)
        Slave to Master Latency: 4380 us (0x00111c)
        Master to Slave PHY: 0x02
          LE 2M
        Slave to Master PHY: 0x02
          LE 2M
        Number of Subevents: 3
        Master to Slave Burst Number: 1
        Slave to Master Burst Number: 0
        Master to Slave Flush Timeout: 1
        Slave to Master Flush Timeout: 1
        Master to Slave MTU: 40
        Slave to Master MTU: 0
        ISO Interval: 8
< HCI Command: LE Setup Isochronous Data Path (0x08|0x006e) plen 13                                                                                                                                                                            #156 31.010900
        Handle: 2
        Data Path Direction: Output (Controller to Host) (0x01)
        Data Path: HCI (0x00)
        Coding Format: u-law log (0x00)
        Company Codec ID: Ericsson Technology Licensing (0)
        Vendor Codec ID: 0
        Controller Delay: 0 us (0x000000)
        Codec Configuration Length: 0
        Codec Configuration: 
> HCI Event: Command Complete (0x0e) plen 6                                                                                                                                                                                                    #157 31.011100
      LE Setup Isochronous Data Path (0x08|0x006e) ncmd 1
        Status: Success (0x00)
        Connection handle: 2
< ACL Data TX: Handle 0 flags 0x00 dlen 16                                                                                                                                                                                                     #158 31.011300
      ATT: Handle Value Notification (0x1b) len 11
        Handle: 0x0012
          Data: 010400000403020200
        ASCS: Audio Stream Source Endpoint (Handle: 0x0012)
        ASE ID: 1
        Streaming
          CIG: 0
          CIS: 0
          Metadata: 0x03020200
> HCI Event: Number of Completed Packets (0x13) plen 5                                                                                                                                                                                         #159 31.073900
        Num handles: 1
        Handle: 0
        Count: 1
> HCI Event: LE Meta Event (0x3e) plen 7                                                                                                                                                                                                       #160 31.074200
      LE Connected Isochronous Stream Request (0x1a)
        ACL Handle: 0
        CIS Handle: 3
        CIG ID: 0x00
        CIS ID: 0x01
< HCI Command: LE Accept Connected Isochronous Stream Request (0x08|0x0066) plen 2                                                                                                                                                             #161 31.074300
        CIS Handle: 3
> HCI Event: Command Status (0x0f) plen 4                                                                                                                                                                                                      #162 31.074500
      LE Accept Connected Isochronous Stream Request (0x08|0x0066) ncmd 1
        Status: Success (0x00)
> HCI Event: LE Meta Event (0x3e) plen 29                                                                                                                                                                                                      #163 31.308100
      LE Connected Isochronous Stream Established (0x19)
        Status: Success (0x00)
        Connection Handle: 3
        CIG Synchronization Delay: 4380 us (0x00111c)
        CIS Synchronization Delay: 2190 us (0x00088e)
        Master to Slave Latency: 4380 us (0x00111c)
        Slave to Master Latency: 4380 us (0x00111c)
        Master to Slave PHY: 0x02
          LE 2M
        Slave to Master PHY: 0x02
          LE 2M
        Number of Subevents: 3
        Master to Slave Burst Number: 1
        Slave to Master Burst Number: 0
        Master to Slave Flush Timeout: 1
        Slave to Master Flush Timeout: 1
        Master to Slave MTU: 40
        Slave to Master MTU: 0
        ISO Interval: 8
```

```
[00:00:02.265,300] <err> bt_audio_stream: iso_qos 0x20003fa4, io 0x200040c0, qos 0x20000194
[00:00:02.265,300] <err> bt_audio_stream: iso_qos 0x20003fc8, io 0x200041b0, qos 0x20000194
[00:00:03.271,000] <err> os: ***** MPU FAULT *****
[00:00:03.271,000] <err> os:   Instruction Access Violation
[00:00:03.271,000] <err> os: r0/a1:  0x20001880  r1/a2:  0x00000000  r2/a3:  0x00000000
[00:00:03.271,000] <err> os: r3/a4:  0x00000000 r12/ip:  0x00000000 r14/lr:  0x0000dacd
[00:00:03.271,000] <err> os:  xpsr:  0x60100000
[00:00:03.271,000] <err> os: s[ 0]:  0x00000000  s[ 1]:  0x00000000  s[ 2]:  0x00000000  s[ 3]:  0x00000000
[00:00:03.271,100] <err> os: s[ 4]:  0x00000000  s[ 5]:  0x00000000  s[ 6]:  0x00000000  s[ 7]:  0x00000000
[00:00:03.271,100] <err> os: s[ 8]:  0x00000000  s[ 9]:  0x00000000  s[10]:  0x00000000  s[11]:  0x00000000
[00:00:03.271,100] <err> os: s[12]:  0x00000000  s[13]:  0x00000000  s[14]:  0x00000000  s[15]:  0x00000000
[00:00:03.271,100] <err> os: fpscr:  0x20003c04
[00:00:03.271,100] <err> os: Faulting instruction address (r15/pc): 0x20006608
[00:00:03.271,100] <err> os: >>> ZEPHYR FATAL ERROR 0: CPU exception on CPU 0
[00:00:03.271,200] <err> os: Current thread: 0x20001c80 (unknown)
[00:00:03.406,400] <err> os: Halting system
```

HCI logs:
[46369.zip](https://github.com/zephyrproject-rtos/zephyr/files/8860464/46369.zip)

### PR DESCRIPTION
**Pull request by Thalley**
Fixes issues for setting up bidirectional unicast audio streams. 

There may be additional issues not covered by this PR (likely caused by some bad merge conflict resolutions). Any additional issues will be fixed in future PRs, so that we can get back to have the samples working right now. 

fixes https://github.com/zephyrproject-rtos/zephyr/issues/46369
fixes https://github.com/zephyrproject-rtos/zephyr/issues/46346

### CODE REVIEW

File: samples/bluetooth/unicast_audio_client/src/main.c [hunk 1]
```diff
@@ -157,22 +158,27 @@ static void lc3_audio_timer_timeout(struct k_work *work)
...
 		for (size_t i = 0U; i < configured_sink_stream_count; i++) {
+			struct net_buf *buf_to_send;
 			int ret;
 
-			ret = bt_audio_stream_send(&streams[i], net_buf_ref(buf));
+			/* Clone the buffer if sending on more than 1 stream */
+			if (i == configured_sink_stream_count - 1) {
+				buf_to_send = buf;
+			} else {
+				buf_to_send = net_buf_clone(buf, K_FOREVER);
```
Comment (Casper-Bonde-Bose) [inline samples/bluetooth/unicast_audio_client/src/main.c:168]: Nice - dual mono!

**Review by Casper-Bonde-Bose — DISMISSED** [review]
LGTM

File: samples/bluetooth/unicast_audio_client/src/main.c (hunk 1 above)
Comment (Thalley) [reply samples/bluetooth/unicast_audio_client/src/main.c:168]: :D 

We can always improve on this in the future if we want to send separate data to each side :)

**Review by asbjornsabo — APPROVED** [review]
LGTM

**Review by sjanc — APPROVED** [review]

**Pull request merged**